src/types/action.types.ts

```diff
@@ -8,6 +8,7 @@ export type ActionBase<
   OnClickArgs extends Array<unknown>,
   OnClickReturnType = void,
 > = {
+  isDisabled?: boolean;
```

Comment:
You shouldn't be extending the generic ActionBase because suddenly all component using this type will be allowed to have `isDisabled` property even if the component itself is not using it. ActionKinds allowes to pass additional props in the third generic argument ActionWithSubactions does not yet so that will needs to be adjusted. But technically you should change the type in https://github.com/securityscorecard/design-system/blob/main/src/components/Datatable/BatchModule/BatchActions/BatchActions.types.ts